### ISSUE TITLE
Password Manager hosted on FreedomBox 

### ISSUE THREAD

**Issue by JosephKiranBabu** [freedombox/Plinth#1050]
Bitwarden has [open-sourced](https://blog.bitwarden.com/host-your-own-open-source-password-manager-ace147649936) their password manager software today with a GPL v3 License.

Cloud-based password managers like Lastpass have always made me feel insecure about a remote data breach that happens out of my control. Yahoo's [data breach in 2013](https://www.wsj.com/articles/yahoo-triples-estimate-of-breached-accounts-to-3-billion-1507062804) cost 3 Billion user accounts.  Like most cloud services, these password managers are honeypots of data for hackers because of their centralized nature.

Let's think of a Password Manager solution hosted on the FreedomBox and come up with a good solution. 
I can currently think of the following solutions:
1. Use a self-hosted password manager like Bitwarden on FreedomBox.
2. Use a file-based solution like KeyPassX store and synchronize it using an encrypted sync tool like Syncthing.

There is a serverless password manager called [Master Password](http://masterpasswordapp.com/), which I'm not using myself but probably should.

**Comment by mray** [comment]
Master Password seems nice but fall short in case you need to *change* your password. Do you have experience with using Bitwarden?

**Comment by JosephKiranBabu** [comment]
@mray I haven't used it yet. It only caught my attention since it was released under GPLv3. I will try it out sometime soon. 

We need to do some brainstorming and research to find the ideal password manager candidate. Let's keep this issue as a discussion thread.

**Comment by JosephKiranBabu** [comment]
Bitwarden is a .Net application. 
I'm hesitant to introduce .Net applications on the FreedomBox after seeing this issue on .Net core.

.NET core should not SPY on users by default https://github.com/dotnet/cli/issues/3093 

We can still turn off telemetry though since it's an automated installation. 
Makes me realize now that to be suitable for FreedomBox it's not enough that the software is FOSS, it has to be privacy-respecting too. This really is a weird exception.

**Comment by JosephKiranBabu** [comment]
An OpenBSD developer is re-implementing the Bitwarden Server in Ruby + Sinatra.
https://github.com/jcs/bitwarden-ruby

**Comment by mray** [comment]
While this sounds totally great the project appears to have a high [bus-factor](https://github.com/jcs/bitwarden-ruby/graphs/contributors).

**Comment by JosephKiranBabu** [comment]
@mray Yeah, the bus factor is a real concern. We had the same problem with Filetea.
